### PR TITLE
Go back to orderBy

### PR DESCRIPTION
**Pull request by forki**

### CODE REVIEW

File: src/fsharp/CompileOps.fs [hunk 1]
```diff
@@ -5102,14 +5102,11 @@ type RootImpls = Zset<QualifiedNameOfFile >
...
-            try
-                match compare q1.Text q2.Text with
-                | 0 -> compare q1.Range.FileName q2.Range.FileName
-                | i -> i
-            with
-            | _ -> -1 }
+    Order.orderBy (fun (q:QualifiedNameOfFile) -> 
+        let fileName = q.Range.FileName
+        if String.IsNullOrWhiteSpace fileName then q.Text else
+        let res = Path.ChangeExtension(Path.GetFullPath(q.Range.FileName).Replace("\\", "/"), "")
```
Comment (alfonsogarciacaro) [inline src/fsharp/CompileOps.fs]: This was more or less my first commit (I think I got the same tests failing as original PR), but I used `null` instead of `""`. I have noticed that `Path.ChangeExtension("foo.fs", "")` gives you `foo.` while `Path.ChangeExtension("foo.fs", null)` gives you `foo` (yeah, I know). Not sure if it makes any difference though.

Comment (forki) [inline src/fsharp/CompileOps.fs]: I also added `if String.IsNullOrWhiteSpace fileName then q.Text else`

Comment (alfonsogarciacaro) [inline src/fsharp/CompileOps.fs]: Yes, that was in vaskir's PR too :)

Comment (alfonsogarciacaro) [inline src/fsharp/CompileOps.fs]: Ah, no, sorry. You added an extra `String.IsNullOrWhiteSpace` on top, I didn't notice ;)
